### ISSUE TITLE
all distributed models produce the same loss per each iteration

### ISSUE THREAD

**Issue by dedoogong** [NVIDIA/apex#840]
Hello, I'm trying to use Apex to train my model with bigger batch size.
I'm using 8 gpus of V100. 
even though I set shuffle=True in the argument of DataLoader, the printed loss was like below;
38651.69
38651.69
38651.69
38651.69
38651.69
38651.69
38651.69
38651.69
58619.0 
58619.0 
58619.0 
58619.0 
58619.0 
58619.0 
58619.0 
58619.0 
34109.03
34109.03
34109.03
34109.03
34109.03
34109.03
34109.03
34109.03
49026.97
49026.97
49026.97
49026.97
49026.97
49026.97
49026.97
49026.97
31109.71
31109.71
31109.71
31109.71
31109.71
31109.71
31109.71
31109.71
39532.15
39532.15
39532.15
39532.15
39532.15
39532.15
39532.15
39532.15
29306.19
29306.19
29306.19
29306.19
29306.19
29306.19
29306.19
29306.19
...

I don't understand why all models produce the same loss value per each iteration when I set the shuffle to True?

here is my command to train my model:
python3.6 -m torch.distributed.launch --nproc_per_node=8 train.py 

and train code looks like:

import torch.distributed as dist
from apex.parallel import DistributedDataParallel as DDP
…
…
parser.add_argument("--local_rank", default=0, type=int) 
opt = parser.parse_args()
opt.distributed = False
if 'WORLD_SIZE' in os.environ:
    opt.distributed = int(os.environ['WORLD_SIZE']) > 1
if opt.distributed: # Set the device according to local_rank.
    torch.cuda.set_device(opt.local_rank)
…
…
torch.distributed.init_process_group(backend='nccl', init_method='env://')
model = DDP(model, delay_allreduce=True)
dataloader = torch.utils.data.DataLoader(
    dataset,
    batch_size=batch_size,
    shuffle=True, #False,
    num_workers=conf['SOLVER']['NUM_WORKER'],
    pin_memory=True,
    collate_fn=dataset.collate_fn,
)
…
…

please give me any useful hint 

Thank you!

**Comment by dedoogong** [comment]
sorry I made a trivial mistake, not adding distributed data sampler ;;
